streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -221,7 +245,23 @@ public StreamsConfig(Map<?, ?> props) {
     }
 
     public Map<String, Object> getConsumerConfigs(StreamThread streamThread, String groupId, String clientId) {
-        Map<String, Object> props = getBaseConsumerConfigs();
+        // filter out non-consumer configs
+        Map<String, ConfigKey> configKeys = ConsumerConfig.configKeys();
+        Map<String, Object> props = filter(configKeys, this.originals());
+
+        // disable auto commit ignoring any user overridden values,
+        // this is necessary for streams commit semantics
+        if (props.containsKey(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG)) {
+            log.warn("Ignoring user specified consumer config " + ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG
+                    + " and always turn off auto committing.");
```

Comment:
Can you please explain why it's better to warn than to fail-fast in this case? Just want to make sure I understand the reasoning for the choice.